packages/path_provider/lib/path_provider.dart

```diff
@@ -21,29 +22,46 @@ const MethodChannel _channel =
 ///
 /// On Android, this uses the `getCacheDir` API on the context.
 Future<Directory> getTemporaryDirectory() async {
-  // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
-  // https://github.com/flutter/flutter/issues/26431
-  // ignore: strong_mode_implicit_dynamic_method
-  final String path = await _channel.invokeMethod('getTemporaryDirectory');
+  final String path =
+      await _channel.invokeMethod<String>('getTemporaryDirectory');
+  if (path == null) {
+    return null;
+  }
+  return Directory(path);
+}
+
+/// Path to a directory where the application may place application support
+/// files.
+///
+/// Use this for files you don’t want exposed to the user. Your app should not
+/// use this directory for user data files.
+///
+/// On iOS, this uses the `NSApplicationSupportDirectory` API.
+/// This directory is not guaranteed to exist and you can call `createSync` to
+/// create it if it doesn't exist.
+///
+/// On Android, this uses the `getDataDirectory` API on the context.
+Future<Directory> getApplicationSupportDirectory() async {
+  final String path =
+      await _channel.invokeMethod<String>('getApplicationSupportDirectory');
   if (path == null) {
     return null;
   }
   return Directory(path);
 }
 
-/// Path to a directory where the application may place files that are private
-/// to the application and will only be cleared when the application itself
-/// is deleted.
+/// Path to a directory where the application may place data that is
+/// user-generated, or that cannot otherwise be recreated by your application.
 ///
-/// On iOS, this uses the `NSDocumentsDirectory` API.
+/// On iOS, this uses the `NSDocumentDirectory` API. Consider using
+/// [getApplicationSupportDirectory] instead if the data is not user-generated.
 ///
-/// On Android, this returns the AppData directory.
+/// On Android, this uses the `getDataDirectory` API on the context. Consider
```

Comment:
(May be out of scope for this PR): It seems like this method is supposed to be used differently on iOS vs Android in a kind of confusing way:

* iOS: use for user generated data
* Android: use for any persistent data that should be invisible to the user

Those use cases aren't really correlated. I feel like they shouldn't really be bundled into the same API if they're really supposed to be used this way, they're different enough that just casing off of Platform version to try and use these correctly would be difficult. I may be reading the docs wrong?

Reply:
I may not be articulating the use case for these paths very effectively in the docs. I'm ok with making the Android and iOS methods separate rather than trying to shoehorn the two platforms onto an imperfect lowest common denominator.